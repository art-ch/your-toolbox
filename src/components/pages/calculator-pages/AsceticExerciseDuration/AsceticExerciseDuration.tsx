'use client';

import { LuCalculator } from 'react-icons/lu';
import { TORSUNOV } from '@/constants/authorMentions';
import { ExerciseTypeTabs } from './components/ExerciseTypeTabs/ExerciseTypeTabs';
import { ToolPageHeading } from '@/components/PageHeading/PageHeading';
import { useTranslation } from 'react-i18next';

export const AsceticExerciseDuration = () => {
  const { t } = useTranslation('asceticExerciseDuration');

  return (
    <div>
      <ToolPageHeading
        icon={<LuCalculator />}
        title={t('title')}
        authorMention={TORSUNOV}
      />
      <h2 className="text-center text-lg sm:text-xl pb-2 font-bold">
        {t('exerciseType')}
      </h2>
      <ExerciseTypeTabs />
    </div>
  );
};
