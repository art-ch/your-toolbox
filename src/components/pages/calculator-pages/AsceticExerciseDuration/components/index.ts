export * from './forms/dynamic-exercise-forms';
export * from './forms/static-exercise-forms';
