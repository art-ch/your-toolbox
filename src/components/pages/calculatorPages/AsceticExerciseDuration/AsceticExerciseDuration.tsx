'use client';

import { ToolPageHeading } from '@/app/components/PageHeading/PageHeading';
import { LuCalculator } from 'react-icons/lu';
import { TORSUNOV } from '@/constants/authorMentions';
import { ExerciseTypeTabs } from './components/ExerciseTypeTabs/ExerciseTypeTabs';

export const AsceticExerciseDuration = () => (
  <div>
    <ToolPageHeading
      icon={<LuCalculator />}
      title="Ascetic Exercise Duration"
      authorMention={TORSUNOV}
    />
    <h2 className="text-center text-lg sm:text-xl pb-2 font-bold">
      Exercise type
    </h2>
    <ExerciseTypeTabs />
  </div>
);
