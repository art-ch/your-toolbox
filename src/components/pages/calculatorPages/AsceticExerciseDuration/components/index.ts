export * from './DynamicExerciseForms';
export * from './StaticExerciseForms';
