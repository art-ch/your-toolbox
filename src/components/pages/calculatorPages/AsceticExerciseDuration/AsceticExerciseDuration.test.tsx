import { render } from '@testing-library/react';
import { AsceticExerciseDuration } from './AsceticExerciseDuration';

describe('AsceticExerciseDuration component', () => {
  it('should render correctly', () => {
    const { container } = render(<AsceticExerciseDuration />);

    expect(container).toMatchSnapshot();
  });
});
