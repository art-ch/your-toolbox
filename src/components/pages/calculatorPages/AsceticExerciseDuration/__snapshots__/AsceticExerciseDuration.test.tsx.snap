// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsceticExerciseDuration component should render correctly 1`] = `
<div>
  <div>
    <header
      class="flex gap-2 md:gap-4 justify-center py-3"
    >
      <div
        class="text-xl sm:text-2xl md:text-4xl pt-0.5"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="20"
            rx="2"
            width="16"
            x="4"
            y="2"
          />
          <line
            x1="8"
            x2="16"
            y1="6"
            y2="6"
          />
          <line
            x1="16"
            x2="16"
            y1="14"
            y2="18"
          />
          <path
            d="M16 10h.01"
          />
          <path
            d="M12 10h.01"
          />
          <path
            d="M8 10h.01"
          />
          <path
            d="M12 14h.01"
          />
          <path
            d="M8 14h.01"
          />
          <path
            d="M12 18h.01"
          />
          <path
            d="M8 18h.01"
          />
        </svg>
      </div>
      <h1
        class="md:text-4xl font-semibold tracking-tight text-xl sm:text-2xl"
      >
        Ascetic Exercise Duration
         
        by
         
        <a
          class="text-blue-500 hover:text-blue-400 focus:text-blue-600"
          href="https://torsunov.ru"
        >
          O.G. Torsunov
        </a>
      </h1>
    </header>
    <section
      class="flex flex-col gap-y-8 py-2 sm:py-4 md:py-6"
    >
      <article>
        <div
          class="lg:pb-4"
        >
          <h2
            class="text-xl md:text-2xl font-semibold leading-none tracking-tight pb-2 sm:pb-0"
          >
            Calculate Cleaned Mental Layers
          </h2>
          <div>
            Calculate how many mental layers you will have cleaned if you walk or run at a set speed for a certain amount of time
          </div>
        </div>
        <div
          class="flex w-full flex-col gap-4 lg:flex-row lg:gap-6"
        >
          <form
            class="lg:min-w-[50%]"
          >
            <div
              class="py-3 sm:py-2 flex flex-col gap-4"
            >
              <div
                class="grid gap-2"
                data-slot="form-item"
              >
                <label
                  class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
                  data-error="false"
                  data-slot="form-label"
                  for="«r0»-form-item"
                >
                  Speed (km/h):
                </label>
                <input
                  aria-describedby="«r0»-form-item-description"
                  aria-invalid="false"
                  class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-hidden focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
                  data-slot="form-control"
                  id="«r0»-form-item"
                  name="speed"
                  placeholder="Please enter speed value"
                  type="number"
                  value=""
                />
                <p
                  class="text-muted-foreground text-sm"
                  data-slot="form-description"
                  id="«r0»-form-item-description"
                >
                  Your walking or running speed
                </p>
              </div>
              <div
                class="grid gap-2"
                data-slot="form-item"
              >
                <label
                  class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
                  data-error="false"
                  data-slot="form-label"
                  for="«r1»-form-item"
                >
                  Duration (minutes):
                </label>
                <input
                  aria-describedby="«r1»-form-item-description"
                  aria-invalid="false"
                  class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-hidden focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
                  data-slot="form-control"
                  id="«r1»-form-item"
                  name="duration"
                  placeholder="Please enter duration value"
                  type="number"
                  value=""
                />
                <p
                  class="text-muted-foreground text-sm"
                  data-slot="form-description"
                  id="«r1»-form-item-description"
                >
                  How many minutes you plan to walk or run
                </p>
              </div>
            </div>
            <button
              class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-8 rounded-md gap-1.5 px-3 has-[>svg]:px-2.5 cursor-pointer"
              data-slot="button"
              disabled=""
              type="submit"
            >
              Calculate
            </button>
          </form>
        </div>
      </article>
      <article>
        <div
          class="lg:pb-4"
        >
          <h2
            class="text-xl md:text-2xl font-semibold leading-none tracking-tight pb-2 sm:pb-0"
          >
            Calculate Speed
          </h2>
          <div>
            Calculate the speed required to clean a specified number of mental layers while walking or running for a specified amount of time.
          </div>
        </div>
        <div
          class="flex w-full flex-col gap-4 lg:flex-row lg:gap-6"
        >
          <form
            class="lg:min-w-[50%]"
          >
            <div
              class="py-3 sm:py-2 flex flex-col gap-4"
            >
              <div
                class="grid gap-2"
                data-slot="form-item"
              >
                <label
                  class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
                  data-error="false"
                  data-slot="form-label"
                  for="«r2»-form-item"
                >
                  Mental layers:
                </label>
                <input
                  aria-describedby="«r2»-form-item-description"
                  aria-invalid="false"
                  class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-hidden focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
                  data-slot="form-control"
                  id="«r2»-form-item"
                  name="cleansingCycles"
                  placeholder="Please enter mental layers value"
                  type="number"
                  value=""
                />
                <p
                  class="text-muted-foreground text-sm"
                  data-slot="form-description"
                  id="«r2»-form-item-description"
                >
                  How many mental layers you want to clean
                </p>
              </div>
              <div
                class="grid gap-2"
                data-slot="form-item"
              >
                <label
                  class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
                  data-error="false"
                  data-slot="form-label"
                  for="«r3»-form-item"
                >
                  Duration (minutes):
                </label>
                <input
                  aria-describedby="«r3»-form-item-description"
                  aria-invalid="false"
                  class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-hidden focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
                  data-slot="form-control"
                  id="«r3»-form-item"
                  name="duration"
                  placeholder="Please enter duration value"
                  type="number"
                  value=""
                />
                <p
                  class="text-muted-foreground text-sm"
                  data-slot="form-description"
                  id="«r3»-form-item-description"
                >
                  How many minutes you plan to walk or run
                </p>
              </div>
            </div>
            <button
              class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-8 rounded-md gap-1.5 px-3 has-[>svg]:px-2.5 cursor-pointer"
              data-slot="button"
              disabled=""
              type="submit"
            >
              Calculate
            </button>
          </form>
        </div>
      </article>
      <article>
        <div
          class="lg:pb-4"
        >
          <h2
            class="text-xl md:text-2xl font-semibold leading-none tracking-tight pb-2 sm:pb-0"
          >
            Calculate Total Time
          </h2>
          <div>
            Calculate the time required to clean a specified number of mental layers while walking or running at a specified speed.
          </div>
        </div>
        <div
          class="flex w-full flex-col gap-4 lg:flex-row lg:gap-6"
        >
          <form
            class="lg:min-w-[50%]"
          >
            <div
              class="py-3 sm:py-2 flex flex-col gap-4"
            >
              <div
                class="grid gap-2"
                data-slot="form-item"
              >
                <label
                  class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
                  data-error="false"
                  data-slot="form-label"
                  for="«r4»-form-item"
                >
                  Mental layers:
                </label>
                <input
                  aria-describedby="«r4»-form-item-description"
                  aria-invalid="false"
                  class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-hidden focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
                  data-slot="form-control"
                  id="«r4»-form-item"
                  name="cleansingCycles"
                  placeholder="Please enter mental layers value"
                  type="number"
                  value=""
                />
                <p
                  class="text-muted-foreground text-sm"
                  data-slot="form-description"
                  id="«r4»-form-item-description"
                >
                  How many mental layers you want to clean
                </p>
              </div>
              <div
                class="grid gap-2"
                data-slot="form-item"
              >
                <label
                  class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
                  data-error="false"
                  data-slot="form-label"
                  for="«r5»-form-item"
                >
                  Speed (km/h):
                </label>
                <input
                  aria-describedby="«r5»-form-item-description"
                  aria-invalid="false"
                  class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-hidden focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm"
                  data-slot="form-control"
                  id="«r5»-form-item"
                  name="speed"
                  placeholder="Please enter speed value"
                  type="number"
                  value=""
                />
                <p
                  class="text-muted-foreground text-sm"
                  data-slot="form-description"
                  id="«r5»-form-item-description"
                >
                  Your walking or running speed
                </p>
              </div>
            </div>
            <button
              class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive bg-primary text-primary-foreground shadow-xs hover:bg-primary/90 h-8 rounded-md gap-1.5 px-3 has-[>svg]:px-2.5 cursor-pointer"
              data-slot="button"
              disabled=""
              type="submit"
            >
              Calculate
            </button>
          </form>
        </div>
      </article>
    </section>
  </div>
</div>
`;
